scripts/BuildLibsRelease.sh

```diff
@@ -116,41 +160,102 @@ lipo -create -output /opt/local/lib/libavformat.a ./libavformat/libavformat.a ./
 lipo -create -output /opt/local/lib/libavdevice.a ./libavdevice/libavdevice.a ./x86_64/libavdevice.a
 lipo -create -output /opt/local/lib/libswresample.a ./libswresample/libswresample.a ./x86_64/libswresample.a
 lipo -create -output /opt/local/lib/libswscale.a ./libswscale/libswscale.a ./x86_64/libswscale.a
+make clean || true    # can exit nonzero
+git status --ignored -s  | colrm 1 2 | xargs rm  -rf
+./configure --disable-asm --disable-x86asm --enable-static --disable-shared --disable-securetransport --extra-cflags="${OSX_VERSION_MIN}" --disable-indev=lavfi --disable-libx264 --disable-lzma --enable-gpl --enable-opengl --disable-programs --disable-optimizations
+sed -i -e "s/^CFLAGS=/CFLAGS=-g ${XL_TARGETS} ${OSX_VERSION_MIN} -DGL_SILENCE_DEPRECATION=1 -Wno-incompatible-function-pointer-types /" ffbuild/config.mak
+sed -i -e "s/^CXXFLAGS=/CXXFLAGS=-g ${XL_TARGETS} ${OSX_VERSION_MIN} -DGL_SILENCE_DEPRECATION=1 /" ffbuild/config.mak
+sed -i -e "s/^LDFLAGS=/LDFLAGS=-g ${XL_TARGETS} ${OSX_VERSION_MIN} /" ffbuild/config.mak
+make -j $NUM_CPUS ; make
+find . -name "*.a" -exec cp {} /opt/local/libdbg/ \;
+popd
 
-echo "libxslwriter"
-git clone https://github.com/jmcnamara/libxlsxwriter.git
-cd libxlsxwriter/cmake
-export CXXFLAGS="-g -O3 -flto=thin ${XL_TARGETS} ${OSX_VERSION_MIN}"
-export CFLAGS="-g -O3 -flto=thin ${XL_TARGETS} ${OSX_VERSION_MIN}"
-export LDFLAGS="-g -O3 -flto=thin ${XL_TARGETS} ${OSX_VERSION_MIN}"
+echo "libxslxwriter"
+if [ ! -d "libxlsxwriter" ] ; then git clone https://github.com/jmcnamara/libxlsxwriter.git ; fi
+pushd libxlsxwriter
+git checkout v1.1.9
+cd cmake
+export CXXFLAGS="-g -O3 -flto=thin ${XL_TARGETS} ${OSX_VERSION_MIN} "
+export CFLAGS="-g -O3 -flto=thin ${XL_TARGETS} ${OSX_VERSION_MIN} "
+export LDFLAGS="-g -O3 -flto=thin ${XL_TARGETS} ${OSX_VERSION_MIN} "
 cmake -DCMAKE_OSX_ARCHITECTURES="x86_64;arm64" ..
 make
 cp libxlsxwriter.a /opt/local/lib/
 git status --ignored -s | colrm 1 3 | xargs rm -rf
+export CXXFLAGS="-g ${XL_TARGETS} ${OSX_VERSION_MIN} "
+export CFLAGS="-g ${XL_TARGETS} ${OSX_VERSION_MIN} "
+export LDFLAGS="-g ${XL_TARGETS} ${OSX_VERSION_MIN} "
+cmake -DCMAKE_OSX_ARCHITECTURES="x86_64;arm64" ..
+make
+cp libxlsxwriter.a /opt/local/libdbg/
+cp -a ../include/* /opt/local/include
 make  clean
 git status --ignored -s | colrm 1 3 | xargs rm -rf
+unset CXXFLAGS
+unset CFLAGS
+unset LDFLAGS
+popd
 
-echo "liblua" #(currently using v5.4.4)
-git clone https://github.com/lua/lua
-cd lua
-git checkout v5.4.4
+
+echo "liblua"   # currently using v5.4.7
+if [ ! -d "liblua" ] ; then git clone https://github.com/lua/lua ; fi
+pushd lua
+git fetch -v
+git checkout v5.4.7
 make clean
-export CFLAGS="-g -O3 -flto=thin -Wall -fno-stack-protector -fno-common ${XL_TARGETS} ${OSX_VERSION_MIN}"
-make -j4 all CFLAGS="$CFLAGS" MYLDFLAGS="$CFLAGS"
+export CFLAGS="-g -O3 -flto=thin -Wall -fno-stack-protector -fno-common ${XL_TARGETS} ${OSX_VERSION_MIN} "
+make -j $NUM_CPUS all CFLAGS="$CFLAGS" MYLDFLAGS="$CFLAGS"
```

Comment:
with the large number of builds, it might be more legible debuggable to any of:
- split this into multiple shells scripts that can succeed or fail individually
- use make with per-target exported flags to update the dependent builds and track their success

Reply:
I would be in favor of this however this would be a functional change from the steps in the readme. Would be interested as to other's thoughts here.